include/matrix_operations.hpp

```diff
@@ -73,6 +77,44 @@ Matrix MatrixOp::init(std::vector<std::vector<std::string>> vec) {
     return result;
 }
 
+/// Method to initialize values of a Matrix object using a double
+Matrix MatrixOp::init(double d) {
+    Matrix result;
+    std::vector<std::vector<double>> * vec_d = new std::vector<std::vector<double>>(1);
+    (*vec_d)[0].push_back(d);
+    result.double_mat = *vec_d;
+    result.to_string();
+    return result;
+}
+
+/// Method to initialize values of a Matrix object using a string
+Matrix MatrixOp::init(std::string s) {
+    Matrix result;
+    std::vector<std::vector<std::string>> * vec_s = new std::vector<std::vector<std::string>>(1);
```

Comment:
You have dynamically allocated memory using `new` keyword. This will result in memory leakage as the memory will not be released explicitly by the compiler when the scope of the variable reaches an end. It would be better if you take advantage of the dynamic nature of `std::vector` to your advantage instead.

Reply:
@aniketsharma00411 I have addressed all suggestions here in my latest commit. I was unable to run the unit tests due to an error. There seems to be no 'gtest/gtest.h' file anywhere that I can find, yet each test .hpp includes it at the top. I looked for a 'test/test.h' file as well but could not find it anywhere. Does this file exist in the current repository? I'm not able to compile and run my tests if it does not

Reply:
Run `git submodule update --init` and then you will be able to run unit tests.

Refer to #69 if you want to delve deeper into the problem but it isn't necessary.

Reply:
Also, delete lines 114 and 124 as suggested by CodeFactor.

Reply:
This might be an installation issue of googletest. Try running the following in lib/googletest/ directory
```
mkdir build
cd build
cmake ..
```
And then try compiling again. Let me if it works or not.

Reply:
I installed cmake and tried running those commands but it gave me an error:

```
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:10 (project):
  The CMAKE_C_COMPILER:

    cl

  is not a full path and was not found in the PATH.
```

Reply:
Which OS are you using?

Reply:
Windows

Reply:
This problem will be resolved when #76 is merged. Try it after that.